src/plugins/cloudinary/index.js

```diff
@@ -234,21 +235,35 @@ class CloudinaryContext extends mixin(Playlistable) {
         this.player.poster(src.poster().url());
       }
 
-      _sources = src.generateSources().filter((src) => {
+      _sources = src.generateSources().map((src) => {
         if (src.isAdaptive) {
           let codec = src.type.split('; ')[1] || null;
           if (codec && 'MediaSource' in window) {
-            let typeStr = `video/mp4; ${src.type.split('; ')[1] || ''}`;
-            return MediaSource.isTypeSupported(typeStr);
+            const parts = src.type.split('; ');
+            let typeStr = `video/mp4; ${parts[1] || ''}`;
+            const canPlay = testCanPlayTypeAndTypeSupported(typeStr);
+            if (vjs.browser.IS_ANY_SAFARI) {
+              // work around safari saying it cant play h265
+              src.type = `${parts[0]}; ${codecShorthandTrans('h264')}`;
+            }
+            if (canPlay) {
+              return src;
+            }
           }
         }
-        return true;
+        return src;
       });
+      // filter out undefined
+      _sources.filter(s => s);
```

Comment:
please remove it after replace `src.generateSources().map` with reduce

Reply:
and 'filter' creates a new array. looks like it does not set the new array to any variable.  I do not completely understand what is it for?

Reply:
@e1adn  done